devicetypes/qubino/qubino-dimmer.src/qubino-dimmer.groovy

```diff
@@ -0,0 +1,634 @@
+			createEvent(name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W")
+		}
+	}
+}
+
+private dimmerEvents(physicalgraph.zwave.Command cmd, ep = null) {
+	createDimmerEvents(cmd.value)
+}
+
+private createDimmerEvents(cmdValue){
+	def value = (cmdValue ? "on" : "off")
+	def result = [createEvent(name: "switch", value: value)]
+	if (cmdValue && cmdValue <= 100) {
+		result << createEvent(name: "level", value: cmdValue == 99 ? 100 : cmdValue)
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd, ep = null) {
+	log.info "SensorMultilevelReport: ${cmd}, endpoint: ${ep}"
+	def result = []
+
+	def map = [:]
+	switch (cmd.sensorType) {
+		case 1:
+			map.name = "temperature"
+			def cmdScale = cmd.scale == 1 ? "F" : "C"
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+			map.unit = getTemperatureScale()
+			break
+		default:
+			map.descriptionText = cmd.toString()
+	}
+	log.debug "SensorMultilevelReport, ${map}, ${map.name}, ${map.value}, ${map.unit}"
+	handleChildEvent(map)
+	result << createEvent(map)
+}
+
+def handleChildEvent(map) {
+	def childDni = "${device.deviceNetworkId}:" + 2
+	log.debug "handleChildEvent / find child device: ${childDni}"
+	def childDevice = childDevices.find { it.deviceNetworkId == childDni }
+
+	if(!childDevice) {
+		log.debug "handleChildEvent / creating a child device"
+		def childComponentLabel	= "Qubino Temperature Sensor"
+		def childDthName		= "Child Temperature Sensor"
+		def childDthNamespace 	= "qubino"
+
+		createChildDevice(childDthNamespace, childDthName, childDni, childComponentLabel)
+		childDevice = childDevices.find { it.deviceNetworkId == childDni }
+	}
+	log.debug "handleChildEvent / sending event: ${map} to child: ${childDevice}"
+	childDevice?.sendEvent(map)
+}
+
+def createChildDevice(childDthNamespace, childDthName, childDni, childComponentLabel) {
+	try {
+		log.debug "Creating a child device: ${childDthNamespace}, ${childDthName}, ${childDni}, ${childComponentLabel}"
+		def childDevice = addChildDevice(childDthNamespace, childDthName, childDni, device.hub.id,
+				[
+					completedSetup: true,
+					label: childComponentLabel,
+					isComponent: false
+				])
+		log.debug "createChildDevice: ${childDevice}"
+		childDevice
+	} catch(Exception e) {
+		log.debug "Exception: ${e}"
+	}
+}
+
+def on() {
+	encapCommands([
+			zwave.basicV1.basicSet(value: 0xFF),
+			zwave.switchMultilevelV3.switchMultilevelGet()
+	], 3000)
+}
+
+def off() {
+	encapCommands([
+			zwave.basicV1.basicSet(value: 0x00),
+			zwave.switchMultilevelV3.switchMultilevelGet()
+	], 3000)
+}
+
+def setLevel(value) {
+	log.debug "setLevel >> value: $value"
+	def valueaux = value as Integer
+	def level = Math.max(Math.min(valueaux, 99), 0)
+	encapCommands([
+			zwave.basicV1.basicSet(value: level),
+			zwave.switchMultilevelV3.switchMultilevelGet()
+	], 3000)
+}
+
+def setLevel(value, duration) {
```

Comment:
this and the durationless `setLevel` can easily be combined.

Reply:
Fixed (the code was refactored).